syntax = "proto3";
package to;

option java_package = "com.epocal.epoctest.to";
option java_outer_classname = "SensorInfoTO";
option optimize_for = LITE_RUNTIME;

message SensorInfo {
int32 channelType = 1;              // enum ChannelType
int32 sensorType = 2;               // enum Sensors
int32 sensorDescriptorNumber = 3;   // was uint8_t
double calDelimit = 4;
double sampleDelimit = 5;
double postDelimit = 6;
double extrapolation = 7;
double calWindowSize = 8;
double sampleWindowSize = 9;
double postWindowSize = 10;
double calCurveWeight = 11;
double sampleCurveWeight = 12;
double calConcentration = 13;
double offset = 14;
double slopeFactor = 15;
double CalMeanLowQC = 16;
double CalMeanHighQC = 17;
double CalDriftLowQC = 18;
double CalDriftHighQC = 19;
double CalSecondLowQC = 20;
double CalSecondHighQC = 21;
double CalNoiseHighQC = 22;
double SampleMeanLowQC = 23;
double SampleMeanHighQC = 24;
double SampleDriftLowQC = 25;
double SampleDriftHighQC = 26;
double SampleSecondLowQC = 27;
double SampleSecondHighQC = 28;
double SampleNoiseHighQC = 29;
double PostMeanLowQC = 30;
double PostMeanHighQC = 31;
double PostDriftLowQC = 32;
double PostDriftHighQC = 33;
double PostSecondLowQC = 34;
double PostSecondHighQC = 35;
double PostNoiseHighQC = 36;
double DeltaDriftLowQC = 37;
double DeltaDriftHighQC = 38;
double param1 = 39;
double param2 = 40;
double param3 = 41;
double param4 = 42;
double param5 = 43;
double param6 = 44;
double param7 = 45;
double param8 = 46;
double param9 = 47;
double param10 = 48;
double param11 = 49;
double param12 = 50;
double param13 = 51;
double param14 = 52;
double param15 = 53;
double param16 = 54;
double param17 = 55;
double param18 = 56;
double param19 = 57;
double param20 = 58;
double param21 = 59;
double param22 = 60;
double param23 = 61;
double param24 = 62;
double param25 = 63;
double param26 = 64;
double param27 = 65;
double param28 = 66;
double param29 = 67;
double param30 = 68;
double param31 = 69;
double param32 = 70;
double param33 = 71;
double param34 = 72;
double param35 = 73;
double param36 = 74;
double param37 = 75;
double param38 = 76;
double param39 = 77;
double param40 = 78;
double param41 = 79;
double param42 = 80;
double param43 = 81;
double param44 = 82;
double param45 = 83;
double param46 = 84;
double param47 = 85;
double param48 = 86;
double param49 = 87;
double param50 = 88;
double param51 = 89;
double param52 = 90;
double param53 = 91;
double param54 = 92;
double param55 = 93;
double param56 = 94;
double param57 = 95;
double param58 = 96;
double param59 = 97;
double param60 = 98;
double param61 = 99;
double param62 = 100;
double param63 = 101;
double param64 = 102;
double param65 = 103;
double param66 = 104;
double param67 = 105;
double param68 = 106;
double param69 = 107;
double param70 = 108;
double param71 = 109;
double param72 = 110;
double param73 = 111;
double param74 = 112;
double param75 = 113;
double param76 = 114;
double param77 = 115;
double param78 = 116;
double param79 = 117;
double param80 = 118;
double param81 = 119;
double param82 = 120;
double param83 = 121;
double param84 = 122;
double param85 = 123;
double param86 = 124;
double param87 = 125;
double param88 = 126;
double param89 = 127;
double param90 = 128;
double param91 = 129;
double param92 = 130;
double param93 = 131;
double param94 = 132;
double param95 = 133;
double param96 = 134;
double param97 = 135;
double param98 = 136;
double param99 = 137;
double param100 = 138;
float readerMeanLow = 139;
float readerMeanHigh = 140;
float readerDriftLow = 141;
float readerDriftHigh = 142;
float readerNoiseLow = 143;
float readerNoiseHigh = 144;
double tMinus = 145;
double tPlus = 146;
double postCurvatureWeight = 147;
int32 bloodPointsToSkip = 148;
int32 bloodPointsInWindow = 149;
double bloodNoiseHigh = 150;
int32 aqPointsToSkip = 151;
int32 aqPointsInWindow = 152;
double aqNoiseHigh = 153;
int32 lateBloodPointsToSkip = 154;
int32 lateBloodPointsInWindow = 155;
double lateBloodNoiseHigh = 156;
int32 lateAqPointsToSkip = 157;
int32 lateAqPointsInWindow = 158;
double lateAqNoiseHigh = 159;
double rtPointLimitLow = 160;
double rtPointLimitHigh = 161;
double d1Low = 162;
double d1High = 163;
double p1d2Low = 164;
double p1d2High = 165;
double p2d2Low = 166;
double p2d2High = 167;
double p3d2Low = 168;
double p3d2High = 169;
double A = 170;
double B = 171;
double C = 172;
double D = 173;
double F = 174;
double G = 175;
double TAmbOffset = 176;
double InjectionTimeOffset = 177;
double AgeOffset = 178;
double PowerOffset = 179;
string NeuralNetBlood = 180;
string NeuralNetAQ = 181;
}
